src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -322,40 +348,40 @@ private void storePrevScopeUninitializedVariableData() {
     }
 
     /**
-     * Update current scope data uninitialized variable according to the previous scope data.
+     * Update current scope data uninitialized variable according to the whole scope data.
      * @param prevScopeUnitializedVariableData variable for previous stack of uninitialized
      *     variables
      */
     // -@cs[CyclomaticComplexity] Breaking apart will damage encapsulation.
-    private void updateUninitializedVariables(Deque<DetailAST> prevScopeUnitializedVariableData) {
+    private void updateAllUninitializedVariables(
+            Deque<DetailAST> prevScopeUnitializedVariableData) {
         // Check for only previous scope
-        for (DetailAST variable : prevScopeUnitializedVariableData) {
-            for (ScopeData scopeData : scopeStack) {
-                final FinalVariableCandidate candidate = scopeData.scope.get(variable.getText());
-                DetailAST storedVariable = null;
-                if (candidate != null) {
-                    storedVariable = candidate.variableIdent;
-                }
-                if (storedVariable != null && isSameVariables(storedVariable, variable)
-                        && !scopeData.uninitializedVariables.contains(storedVariable)) {
-                    scopeData.uninitializedVariables.push(variable);
-                }
-            }
-        }
+        updateUninitializedVariables(prevScopeUnitializedVariableData);
         // Check for rest of the scope
-        for (Deque<DetailAST> unitializedVariableData : prevScopeUninitializedVariables) {
-            for (DetailAST variable : unitializedVariableData) {
+        prevScopeUninitializedVariables.forEach(this::updateUninitializedVariables);
+    }
+
+    /**
+     * Update current scope data uninitialized variable according to the specific scope data.
+     * @param scopeUnitializedVariableData variable for specific stack of uninitialized variables
+     */
+    private void updateUninitializedVariables(Deque<DetailAST> scopeUnitializedVariableData) {
+        final Iterator<DetailAST> iterator = currentScopeAssignedVariables.peek().iterator();
+        while (iterator.hasNext()) {
+            final DetailAST assignedVariable = iterator.next();
+            for (DetailAST variable : scopeUnitializedVariableData) {
                 for (ScopeData scopeData : scopeStack) {
                     final FinalVariableCandidate candidate =
-                        scopeData.scope.get(variable.getText());
+                            scopeData.scope.get(variable.getText());
```

Comment:
Done. BTW, why the verification pass when changing the indentation? My Intellij reformat the indentation automatically. There are also many 8 spaces indentation in other lines with assignment like this.